# cottonRef_ZM113
Scripts used to analyze the T2T reference genome of *G. hirsutum* cv. ZM113

* [rDNA_copy_number.sh](rDNA_copy_number.sh): copy number estimation of rDNA
* [GhSat194_copy_number.sh](GhSat194_copy_number.sh): copy number estimation of GhSat194 ([194.fa](194.fa)) based on genome search and short-read coverage depth
* [GWAS.sh](GWAS.sh): GWAS analysis
* [GWAS_phenotype_tidty.R](GWAS_phenotype_tidty.R): R script to process GWAS results

  

**Other Data and Materials Availability**
* The ZM113 genome assembly and annotation were deposited in NCBI under BioProject accession number [PRJNA1039892](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1039892) and [Cotton Breeding Database](http://222.88.152.130:1130/).
* The raw sequencing reads by Oxford Nanopore Technology ultra-long read sequencing, PacBio high-fidelity sequencing, MGI PE150 short-read sequencing, Hi-C, RNA-seq, ChIP-seq and BS-seq are available in NCBI under BioProject [PRJNA1041574](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1041574) and [Cotton Breeding Database](http://222.88.152.130:1130/).
* The raw Bionao data is available in the Genome Sequence Archive of China National Genomics Data Center under [PRJCA029603 accession CRA018637](https://ngdc.cncb.ac.cn/gsa/search?searchTerm=CRA018637).
* Nine *G. hirsutum* reference genomes and two diploid cotton genomes used were downloaded from [CottonGen](https://cottongen.org/).
